src/Cache.cpp

```diff
@@ -46,10 +46,16 @@ static const std::string_view OLM_ACCOUNT_KEY("olm_account");
 static const std::string_view CACHE_FORMAT_VERSION_KEY("cache_format_version");
 static const std::string_view CURRENT_ONLINE_BACKUP_VERSION("current_online_backup_version");
 
+constexpr auto MAX_DBS = 32384UL;
+#if Q_PROCESSOR_WORDSIZE == 8                                                 // 64-bit
```

Comment:
Can you make this a simple >=? I would probably make the checks a simple >= 0 or else, but I guess anything smaller than 4 will have serious trouble and erroring out in that case is probably good.

Reply:
Processor word size times 8 bit words gives the number of bits in the address space.

Technically 36 bits are enough to address 64GiB of memory (2^36), but [it isn't too common](https://en.wikipedia.org/wiki/36-bit_computing), and I doubt nheko will ever be compiled for those. The closest word size would 5 (5*8=40 bits), so I  can make it that way if you prefer:

```cpp
#if Q_PROCESSOR_WORDSIZE >= 5 // 40-bit or more, up to 2^(8*WORDSIZE) words addressable.
// 32 GiB
#elif Q_PROCESSOR_WORDSIZE == 4 // 32 bit
// 1 GiB
#else
#error Not enough virtual address space for the database on this platform.
#fi
```
This would also allow more-than-64-bit platforms to compile.

I initially didn't take these cases into account as, while 32-bit is esoteric nowadays, anything else other than 64 bits is even rarer.

Reply:
Well, I was mostly thinking about platforms with more than 8 byte wordsize, which are now excluded by your macro. I can imagine there to be some arches with 128bit in the future, i.e. just change the first check to >=8.